app/validate_txs.go

```diff
@@ -8,11 +8,44 @@ import (
 	"github.com/cosmos/cosmos-sdk/telemetry"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 
+	square "github.com/celestiaorg/go-square/v2"
 	"github.com/celestiaorg/go-square/v2/tx"
 
 	"github.com/celestiaorg/celestia-app/v4/pkg/appconsts"
 )
 
+// FilteredBuilder filters txs and blobs using a copy of the state and tx validity
+// rules before adding it the square.
+type FilteredBuilder struct {
+	logger   log.Logger
+	ctx      sdk.Context
```

Comment:
I think we could remove these as fields on the struct and pass `ctx` to `Fill(ctx, req.Txs)`, and any other funcs as necessary

Also could derive a logger from ctx on the fly using something like:
```
logger := ctx.Logger().With("app/filtered-square-builder")
```